### PR TITLE
SW-5156 Move Documents, Methodologies, Values, and Variables redux from pdd-web into the app

### PR DESCRIPTION
**Pull request by nickgraz**
These are only hooked up to redux for now, currently there are no views or components that interact with them.

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#2489** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2489?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/redux/rootReducer.ts [hunk 1]
```diff
@@ -79,6 +92,15 @@ export const reducers = {
   deliverablesEdit: deliverablesEditReducer,
   deliverablesSearch: deliverablesSearchReducer,
   deliverables: deliverablesReducer,
+  documentProducerDocumentList: documentListReducer,
```
Comment (karthikbtf) [inline src/redux/rootReducer.ts:95]: Would it be time to export single payloads of reducers from the xyzslice files? We could import the payload and spread that in the rootReducer.
I can take a quick stab at it.

File: src/redux/features/documentProducer/documents/documentsSlice.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ */
+type DocumentRequestsState = Record<string, StatusT<number>>;
+const initialDocumentRequestsState: DocumentRequestsState = {};
+
+export const documentRequestsSlice = createSlice({
+  name: 'documentRequestsSlice',
+  initialState: initialDocumentRequestsState,
+  reducers: {},
+  extraReducers: (builder: ActionReducerMapBuilder<DocumentRequestsState>) => {
+    buildReducers(requestCreateDocument)(builder);
```
Comment (karthikbtf) [inline src/redux/features/documentProducer/documents/documentsSlice.ts:85]: Nice, I understand this was a tad bit different in pdd, appreciate the reconciliation here.

**Review by karthikbtf — APPROVED** [review]
Nice 👍

File: src/redux/rootReducer.ts (hunk 1 above)
Comment (nickgraz) [reply src/redux/rootReducer.ts:95]: Yea that would be cleaner, I can do that now.

File: src/redux/features/documentProducer/index.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import { documentProducerVariablesReducers } from './variables/variablesSlice';
+
+const documentProducerReducers = {
+  ...documentProducerDocumentsReducers,
+  ...documentProducerMethodologiesReducers,
+  ...documentProducerVariablesReducers,
+  ...documentProducerVariableValuesReducers,
+};
+
+export default documentProducerReducers;
```
Comment (karthikbtf) [inline src/redux/features/documentProducer/index.ts:13]: Nice 👍

File: src/redux/rootReducer.ts [hunk 4]
```diff
@@ -79,6 +80,7 @@ export const reducers = {
   deliverablesEdit: deliverablesEditReducer,
   deliverablesSearch: deliverablesSearchReducer,
   deliverables: deliverablesReducer,
+  ...documentProducerReducers,
```
Comment (karthikbtf) [inline src/redux/rootReducer.ts:83]: 💯

**Review by karthikbtf — APPROVED** [review]
Thanks so much!!

File: src/redux/rootReducer.ts (hunk 4 above)
Comment (joshbuchea) [reply src/redux/rootReducer.ts:83]: Nice!!

**Pull request merged**